web/js/codeworld_shared.js

```diff
@@ -532,12 +537,48 @@ function registerStandardHints(successFunc) {
       found[2] = [];
     }
 
-    const options = found[0].concat(found[1]).concat(found[2]);
+    const mappedTerms = {
+      square: 'Prelude.rectangle',
+      triangle: 'Prelude.polygon',
+    };
+
+    for (const [lookedUpTerm, replacementTerm] of Object.entries(mappedTerms)) {
+      let matchingCandidate;
+      for (const bucket of found) {
+        matchingCandidate = bucket.find(({ text }) =>
+          replacementTerm.includes(text)
+        );
+
+        if (matchingCandidate) {
+          break;
+        }
+      }
+
+      if (window.codeWorldSymbols[replacementTerm] && !matchingCandidate) {
```

Comment:
You can check `window.codeWorldSymbols[replacementTerm]` and `continue;` at the beginning of the loop.  It will save work for cases where that's false.